### PR TITLE
Business Match API (closes #6)

### LINKED ISSUE #6: No business match API?
https://www.yelp.com/developers/documentation/v3/business_match

### PR DESCRIPTION
**Pull request by zolrath**
This adds the Business Match API as requested in #6 
I didn't pull request it as the match API didn't seem to be very effective for me in my testing, though it could have simply been the businesses I was trying to match on since the endpoint is beta.

### CODE REVIEW

**Comment by gfairchild** [comment]
Upon a quick inspection, this looks solid. Would you mind adding a bullet at the bottom of `README.md` indicating that we now support this endpoint?

**Comment by zolrath** [comment]
You got it!
The examples in the `examples.py` file aren't great as I haven't actually gotten a reasonable response out of the API just yet. I tend to get no results with `best` and two sometimes unrelated results with lookup.

**Comment by gfairchild** [comment]
Awesome! I haven't had a chance to actually pull this down and test it yet, but I'll do that shortly (most likely tomorrow). Would you mind making that `.get()` change? I think that's the last thing I see offhand.

**Comment by gfairchild** [comment]
Oh, another 2 small changes:

1. Would you mind adding an entry to `CHANGES.md` describing this change? Let's up the version to 2.1.0.
1. Would you mind updating `setup.py` to specify version 2.1.0?

**Comment by zolrath** [comment]
Version bumped!

**Comment by zolrath** [comment]
What's the `.get()` change you requested? I don't see any code comments.

File: CHANGES.md [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
+
 ## 2.0.1 (2017-02-27)
 * Remove reliance on `ez_setup.py`.
 * Add `setup.cfg` to instruct the build process to build universal wheels.
 
 ## 2.0 (2017-02-27)
 * Implement the new Yelp Fusion API (aka Yelp v3 API). Note that yelpapi v1.4 is the last version to support the Yelp v2 API. [The Yelp v2 API is being deprecated, so all new developers should move to the new Fusion API ASAP.](https://engineeringblog.yelp.com/2017/02/recent-improvements-to-the-fusion-api.html)
-  * Use OAuth 2.0 for authentication.
-  * Implement new [Transaction Search API](https://www.yelp.com/developers/documentation/v3/transactions_search), [Reviews API](https://www.yelp.com/developers/documentation/v3/business_reviews), and [Autocomplete API](https://www.yelp.com/developers/documentation/v3/autocomplete).
+* Use OAuth 2.0 for authentication.
```
Comment (gfairchild) [inline CHANGES.md]: I'd rather leave these indented since I was really just calling out specific changes related to the new Fusion API.

**Comment by gfairchild** [comment]
Ah, sorry! I was using GitHub's code review feature and didn't hit "submit". Do you see the comment on `.get()` now?

**Comment by zolrath** [comment]
I've made the requested changes!

**Comment by gfairchild** [comment]
Looks great to me! Merging. Thanks!

**Pull request merged**